app/shell2/code.cpp

```diff
@@ -133,7 +135,17 @@ FullCommandList::FullCommandList(EgalitoInterface *egalito) {
             {"-k", ArgumentSpec({"-k"}, ArgumentSpec::TYPE_FLAG)}
         }, {}, 0),
         [this, egalito] (ShellState &state, ArgumentValueList &args) {
-            std::string output = tempnam("/tmp", "ega-");
+            //std::string output = tempnam("/tmp", "ega-");
+            char pfnam[TPATH_MAX];
+            char tfnam[TPATH_MAX];
+
+            memset(pfnam, 0, TPATH_MAX);
+            memset(tfnam, 0, TPATH_MAX);
+
+            snprintf(pfnam, TPATH_MAX-1, "/proc/self/fd/%d",
```

Comment:
std::snprintf + `#include <cstdio>`
or `#include <stdio.h>`

Reply:
Ack

Reply:
What do you prefer?